bbs_plus/src/proof.rs

```diff
@@ -339,7 +339,7 @@ impl<E: Pairing> PoKOfSignatureG1Protocol<E> {
             params.h[*i].serialize_compressed(&mut writer)?;
             bases_revealed.push(params.h[*i]);
             msg.serialize_compressed(&mut writer)?;
-            exponents.push(msg.clone());
+            exponents.push(*msg);
```

Comment:
`clippy`

Reply:
Is there an impact on performance?

Reply:
Using `*` is shorter and we can understand that this operation is cheap.
Also, `The behavior of Copy is not overloadable; it is always a simple bit-wise copy`, so it might be cheaper if the `clone` implementation does something else.
See discussion here: https://github.com/rust-lang/rust-clippy/issues/2870